device/cuda/src/finding/kernels/specializations/apply_interaction_src.cuh

```diff
@@ -17,12 +17,20 @@
 namespace traccc::cuda::kernels {
 
 template <typename detector_t>
-__global__ void apply_interaction(
+__global__ void _apply_interaction(
```

Comment:
I'm not super happy about this naming. 🤔

My first idea would've been to give some new name to the C\+\+ functions, and keep the name of the kernels the same.

But another option could be to call the actual kernel something like `traccc::cuda::kernels::impl::apply_interaction`. 🤔 

I'm just not a fan of using functions with underscore prefixes. 😦

Reply:
Sure, we can do this in a little `impl` namespace. :+1: